### DISCUSSION TITLE
How to change Edge style based on the output value of the connected Node?

### DISCUSSION THREAD

**Discussion by microdou** [open-source-labs/Svelvet#353] [General]
It may be related to this topic #335, but I'm looking for a specific implementation or **workaround** in current version.

I'd like to automatically set `animate={true}` for an Edge if its upstream Node/Anchor output value (initially default to `null`) is not `null`. In this way, when a Node/Anchor has a computed output value (or meet certain criteria), downstream edges of the Node (or specific output Anchor) are animated, otherwise not animated. the animated Edge is a good way to represent the actual data flow path.

I have tried to use `edge` prop to dynamically load custom edge components to Node / Anchor, but with no luck. The `edge` prop doesn't seem to be reactive. It was possible in `v6`.

**Comment by microdou** [comment]
Here is an example repo:

https://stackblitz.com/edit/svelvet-carbon-xcud5s?file=src%2Froutes%2FNodeSlider%2Findex.svelte

I'd like to animate the connected edge when output value of the Slider Node is > 5. Currently the Edge is not reactive.

```svelte
<!-- AnimatedEdge.svelte -->
<script lang="ts">
  import { Edge } from "svelvet";
</script>

<Edge width={4} animate />
```

```svelte
<!-- Node.svelte -->
<script>
import AnimatedEdge from "./AnimatedEdge.svelte";
// ...
const output = generateOutput(inputs, procesor);
</script>

<Node edge={$output > 5 ? AnimatedEdge : null}>
  <!-- ... -->
</Node>
```

**Comment by briangregoryholmes** [comment]
See discussion here for future plans and how to implement something similar in the current version: https://github.com/open-source-labs/Svelvet/discussions/363

I'll let you know as soon as the new Edge composability is live. It will make this exact use case very simple.

**Reply by microdou** [reply]
@briangregoryholmes Thank you so much! It's apparently already live since commit https://github.com/open-source-labs/Svelvet/commit/954e532dbb6c022a8ec8729a70d495d0eb7082fd. The Edge slot works great so far!
